src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineComputer.java

```diff
@@ -117,9 +123,18 @@ public HttpResponse doDoDelete() throws IOException {
         ComputeEngineInstance node = getNode();
         if (node != null) {
             try {
+                ComputeEngineCloud cloud = getCloud();
+
+                // Checks for failed jobs for this computer's node
+                if (cloud != null && !this.getBuilds().failureOnly().isEmpty()) {
```

Comment:
Are we sure that this `chain` cant throw NPE?

Reply:
If cloud is not found, there is a CloudNotFoundException. However, findBugs wants a null check here.

Reply:
Im talking about this part `!this.getBuilds().failureOnly().isEmpty()`

Reply:
Ah, I see what you mean by chain
https://javadoc.jenkins.io/hudson/model/Job.html#getBuilds-- never returns null
https://javadoc.jenkins.io/hudson/util/RunList.html#failureOnly-- searches through the list, could be empty